### PR TITLE
internal/{v5,charmstore}: fix so charmstore can work with no IdentityAPIURL

### PR DESCRIPTION
**Pull request by rogpeppe**
Quite a few testing clients rely on being able to do this, so let's not break them.

Also make the example config.yaml do the right thing.

### CODE REVIEW

File: internal/charmstore/server_test.go [hunk 1]
```diff
@@ -91,43 +92,68 @@ func (s *ServerSuite) TestNewServerWithVersions(c *gc.C) {
...
-	h, err := NewServer(s.Session.DB("foo"), nil, serverParams, map[string]NewAPIHandlerFunc{
+	h, err := NewServer(s.Session.DB("foo"), nil, params, map[string]NewAPIHandlerFunc{
 		"version1": serveConfig,
 	})
 	c.Assert(err, gc.IsNil)
 	defer h.Close()
+
+	// The IdentityLocation field is filled out from the IdentityAPIURL
+	// and the final slash is trimmed.
+
```
Comment (mhilton) [inline internal/charmstore/server_test.go:116]: Perhaps remove this blank line, the comment seems to be unrelated to anything here.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1760/
Test PASSed.

**Comment by mhilton** [comment]
:+1:

**Comment by fabricematrat** [comment]
:+1:  Note that will will probably need to change the charm.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1761/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1762/
Test PASSed.

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge
